alacritty/src/display.rs

```diff
@@ -381,63 +363,61 @@ impl Display {
-            self.size_info.height = (size.height as f32).max(cell_height + 2. * padding_y);
-        }
-
-        // Distribute excess padding equally on all sides.
-        if config.ui_config.window.dynamic_padding {
-            padding_x = dynamic_padding(padding_x, self.size_info.width, cell_width);
-            padding_y = dynamic_padding(padding_y, self.size_info.height, cell_height);
-        }
-
-        self.size_info.padding_x = padding_x.floor() as f32;
-        self.size_info.padding_y = padding_y.floor() as f32;
-
-        let mut pty_size = self.size_info;
-
-        // Subtract message bar lines from pty size.
-        if let Some(message) = message_buffer.message() {
-            let lines = message.text(&self.size_info).len();
-            pty_size.height -= pty_size.cell_height * lines as f32;
-        }
+        // TODO: Maybe combine, potentiall shorten to one line
+        let width = update_pending
+            .dimensions()
+            .map(|size| size.width as f32)
+            .unwrap_or_else(|| self.size_info.width());
+        let height = update_pending
+            .dimensions()
+            .map(|size| size.height as f32)
+            .unwrap_or_else(|| self.size_info.height());
+
+        self.size_info = SizeInfo::new(
+            width,
+            height,
+            cell_width,
+            cell_height,
+            f32::from(config.ui_config.window.padding.x),
+            f32::from(config.ui_config.window.padding.y),
+            self.size_info.dpr,
+            config.ui_config.window.dynamic_padding,
+        );
 
-        // Add an extra line for the current search regex.
-        if search_active {
-            pty_size.height -= pty_size.cell_height;
-        }
+        // Update number of column/lines in the viewport.
+        let message_bar_lines =
+            message_buffer.message().map(|m| m.text(&self.size_info).len()).unwrap_or(0);
+        let search_lines = if search_active { 1 } else { 0 };
+        self.size_info.reserve_lines(message_bar_lines + search_lines);
 
         // Resize PTY.
-        pty_resize_handle.on_resize(&pty_size);
+        pty_resize_handle.on_resize(&self.size_info);
 
         // Resize terminal.
-        terminal.resize(pty_size);
+        terminal.resize(self.size_info);
 
         // Resize renderer.
-        let physical = PhysicalSize::new(self.size_info.width as u32, self.size_info.height as u32);
+        let physical =
+            PhysicalSize::new(self.size_info.width() as u32, self.size_info.height() as u32);
         self.window.resize(physical);
         self.renderer.resize(&self.size_info);
 
-        info!("Padding: {} x {}", self.size_info.padding_x, self.size_info.padding_y);
-        info!("Width: {}, Height: {}", self.size_info.width, self.size_info.height);
+        info!("Padding: {} x {}", self.size_info.padding_x(), self.size_info.padding_y());
+        info!("Width: {}, Height: {}", self.size_info.width(), self.size_info.height());
```

Comment:
```suggestion
        info!("Width: {}, height: {}", self.size_info.width(), self.size_info.height());
```

Reply:
I much prefer it with a capitalized H actually, the same way it is on master.